builder/azure/arm/step_deploy_template.go

```diff
@@ -165,7 +166,7 @@ func (s *StepDeployTemplate) getImageDetails(ctx context.Context, resourceGroupN
 func deleteResource(ctx context.Context, client *AzureClient, resourceType string, resourceName string, resourceGroupName string) error {
 	switch resourceType {
 	case "Microsoft.Compute/virtualMachines":
-		f, err := client.VirtualMachinesClient.Delete(ctx, resourceGroupName, resourceName)
+		f, err := client.VirtualMachinesClient.Delete(ctx, resourceGroupName, resourceName, flag.Bool("forceDelete", true, "forces fast deletion"))
```

Comment:
```suggestion
		f, err := client.VirtualMachinesClient.Delete(ctx, resourceGroupName, resourceName, false)
```

See https://github.com/hashicorp/packer-plugin-azure/pull/199#issuecomment-1135075590

Reply:
fixed